src/reducers/characterReducer.js

```diff
@@ -181,9 +181,36 @@ const characterReducer = createSlice({
     addCharacter: (state, action) => {
       state.characters.push(action.payload)
     },
+    editCharacter: (state, action) => {
+      const {
+        id,
+        name,
+        eyeColor,
+        height,
+        gender,
+        birthYear,
+        homeworld,
+        species,
+        filmConnection,
+      } = action.payload
+      const existingCharacter = state.characters.find(
+        (existingCharacter) => existingCharacter.id === id
+      )
+
+      if (existingCharacter) {
+        ;(existingCharacter.name = name),
+          (existingCharacter.eyeColor = eyeColor),
+          (existingCharacter.height = height),
+          (existingCharacter.gender = gender),
+          (existingCharacter.birthYear = birthYear),
```

Comment:
Please read about spreading properties in javascript to see how this might be simplified